src/sodiumCompatibility/java/net/coderbot/iris/compat/sodium/impl/vertex_format/xhfp/XHFPModelVertexBufferWriterNio.java

```diff
@@ -16,7 +19,7 @@
     private MaterialIdHolder idHolder;
 
     public XHFPModelVertexBufferWriterNio(VertexBufferView backingBuffer) {
-        super(backingBuffer, IrisModelVertexFormats.MODEL_VERTEX_XHFP);
+        super(backingBuffer, Iris.isPackActive() ? IrisModelVertexFormats.MODEL_VERTEX_XHFP : DefaultModelVertexFormats.MODEL_VERTEX_HFP);
```

Comment:
This probably doesn't need to be changed - this class should only ever be used if the extended vertex format is being actively used.